### DISCUSSION TITLE
Multi-root file organization for different filesystems

### DISCUSSION THREAD

**Discussion by josh-hemphill** [directus/directus#17618] [Feature Requests]
### Summary

When trying to manage different file sources that with different requirements for what and how things get stored, having a way to separate it out would be invaluable.

### Motivation

Managing multiple file driver targets and having to deal with things like file auditing and security requirements; some stuff can be thrown into buckets, others have to be in specific locations, and being able to manage them separately is useful.

### Detailed Design

The admin area could have a multi-item selection for the mounted file targets. Since the file drivers still seem to be the domain of the env/config files, it might make sense to keep it that way and just surface the enumeration of them to the admin settings. It might be worthwhile to use the same driver instance multiple times from the Admin settings if there's also an added way to differentiate where things are stored for each driver so files can be separated on that storage target for auditing/security, if you don't "have" to use a separate driver instance, but that idea would more complicated and not make sense to do right away.

The file management page would work pretty well, just have to duplicate the existing root folder drop down. 🤷‍♂️

When setting the file target for a field, it could remain the same unless there is more than one root configured, but just add the existing file inner dropdown functionality to the root folders.

### Must Have:
- Multiple root directories under the file management page

### Should Have:
- Each relate to a separate driver in some way, or a different implementation 

### Could Have:
- Multiple roots using the same driver instance, but differentiating by admin configuration, like subdirectories or tags

### Drawbacks

There would be some increase to complexity.
The admin settings would be a little more complicated.
The file management would remain essentially the same for anyone using it.
When creating schemas and using file types, selecting a target might be very slightly more complicated only when multiple roots are configured.

### Alternatives

I thought about using folders or something and intercepting with an extension to direct to proper drivers, but it wouldn't be easy to block saving to the root, and the more I think about it, the more impractical it seems.

### Adoption Strategy

I'm not sure. It __*might*__ be non-breaking.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
